agent.py

```diff
@@ -78,9 +83,32 @@ def _handle_sigterm(self, signum, frame):
         log.debug("Collector is stopped.")
 
     def _handle_sigusr1(self, signum, frame):
+        """Handles SIGUSR1, which signals an exit with an autorestart."""
         self._handle_sigterm(signum, frame)
         self._do_restart()
 
+    def _handle_sigusr2(self, signum, frame):
```

Comment:
I guess I would expect `SIGHUP` to implement such behavior in lieu of `SIGUSR2`.